File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.py
def solution(n):
    if n < 2:
        return 0
    
    # 0과 1은 소수가 아니므로 False로 시작
    is_prime = [True] * (n + 1)
    is_prime[0] = is_prime[1] = False

    # 제곱근까지 배수 제거
    for i in range(2, int(n ** 0.5) + 1):
        if is_prime[i]:
            for j in range(i * i, n + 1, i):
                is_prime[j] = False

    return sum(is_prime)

File: Swift/프로그래머스/2/42839. 소수 찾기/소수 찾기.swift
import Foundation

func isPrime(_ n: Int) -> Bool {
    if n < 2 { return false }
    if n == 2 { return true }

    let limit = Int(Double(n).squareRoot())
    if limit < 2 { return true }

    for i in 2...limit {
        if n % i == 0 { return false }
    }
    return true
}

func solution(_ numbers:String) -> Int {
    let digits = Array(numbers)
    var visited = Array(repeating: false, count: digits.count)
    var madeNumbers = Set<Int>()

    func dfs(_ current: String) {
        if !current.isEmpty {
            madeNumbers.insert(Int(current)!)
        }

        for i in 0..<digits.count {
            if visited[i] { continue }

            visited[i] = true
            dfs(current + String(digits[i]))
            visited[i] = false
        }
    }

    dfs("")

    return madeNumbers.filter { isPrime($0) }.count
}